proto/pulumi/provider.proto

```diff
@@ -115,13 +115,23 @@ service ResourceProvider {
     // Call dynamically executes a method in the provider associated with a component resource.
     rpc Call(CallRequest) returns (CallResponse) {}
 
-    // Check validates that the given property bag is valid for a resource of the given type and returns the inputs
-    // that should be passed to successive calls to Diff, Create, or Update for this resource. As a rule, the provider
-    // inputs returned by a call to Check should preserve the original representation of the properties as present in
-    // the program inputs. Though this rule is not required for correctness, violations thereof can negatively impact
-    // the end-user experience, as the provider inputs are using for detecting and rendering diffs.
+    // `Check` validates a set of input properties against a given resource type. A `Check` call returns either a set of
```

Comment:
I truly think that a more accurate and useful explanation would be that Check produces a planned resource state, including defaults and incorporating prior state.  The planned resource state is later compared to the actual state to  determine whether any remedial action is needed.

Please explain that Check receives the olds and what to do with it.